packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -461,11 +462,36 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const previousProps = usePreviousProps({
+    filteredOptions,
+    value,
+  });
+
   const syncHighlightedIndex = React.useCallback(() => {
     if (!popupOpen) {
       return;
     }
 
+    if (
+      highlightedIndexRef.current !== -1 &&
+      previousProps.filteredOptions &&
+      previousProps.filteredOptions.length !== filteredOptions.length &&
+      (Array.isArray(previousProps.value)
+        ? previousProps.value.every((val, i) => value[i] === val)
+        : previousProps.value === value)
```

Comment:
Will reference equality for objects work in this case, or would `inputValue === getOptionLabel(value)` be preferable? I am not sure about this.

Reply:
Updated code to use getOptionLabel to check equality of options when options are provided as objects